### ISSUE TITLE
Tooling stuff

### ISSUE THREAD

**Issue by StoneCypher** [jatlasd/gitchess#10]
# What's the deal?

Here's the thing - you're going to want to install a couple of libraries.  Not many, but, also not zero.

Given how lightweight this app is, you could definitely get away with doing it the old way, just manually putting a couple of .js libs in the distribution and calling them, and then calling it a day.

But sooner or later you're going to need to learn `node` and `npm`, and installing packages is one major way that gets needed.

Here, you don't need much.  Just a couple of static files. 

But later, you'll want to do `react` someday, and ... well at that point you really need installed libs and a bundler.

On the upside, once you've got `node` installed, you can teach it to be your little robot, and reliably do tasks for you.  My setup burns the old build down, runs the checks, runs the tests, remakes the site for me, all automatically.

Also on the upside, this makes testing practical.  (You *can* test without node, but it's more of a pain than just setting node up, and unless you've got not-invented-here syndrome to a truly insane degree, you're not getting things like coverage out of it.)

Also also on the upside, this makes things like GH actions meaningfully usable.

This project is sort of on the knife's edge of warranting this kind of investment.  It's a little over the top, but not much, and it would pay out, and it's something you need to learn eventually - presented in a small safe isolated context.

But it would add a couple days of learning and hard thinking.

Your call.  Either way is fine.

Please either close this with a "no, I just want to finish it as stands," or leave it open and @ tag me with a "yes, I'd like to automate this project."

&nbsp;

## What would tooling this project mean?

* [x] Install `node`
    * [ ] move the site into a `src/site` tree
    * [ ] move the app into a `src/js` tree
* [ ] make a clean and build loop
    * [ ] destroy `docs/` and recreate the directory empty
    * [ ] clone the site and app back into `docs/`
* [ ] Install interact.js
* [ ] Set up a basic test rig
    * [ ] Analyze a few move responses, so that you can see how to test game logic
* [ ] Set up Github Actions

**Comment by jatlasd** [comment]
@StoneCypher yes, I would love to tackle all of this. Anything to get a deeper understanding and begin to take more practical approaches.

**Comment by StoneCypher** [comment]
Let's re-open this then, and I'll use it as a place to build out next steps
